### ISSUE TITLE
Cannot authenticate on Linux

### ISSUE THREAD

**Issue by fbreitwieser** [alangrainger/obsidian-google-photos#20]
Hi, thanks a lot for this great plugin! Looking forward to using it.

I'm currently stuck trying to authenticate. I'm on Linux with Obsidian v1.4 and tried with both Wayland and Xorg.

When I try to authenticate, the window opens asking me to choose an account, and then to give access to view the library. 

However, when I click 'Approve', the window title changes to 'Approved Clicked', the content turns blank, and the popup window stays open. When I manually close it, I get a message 'Authenthication Failed'.

Can you help with this?

**Comment by fbreitwieser** [comment]
![image](https://github.com/alangrainger/obsidian-google-photos/assets/516060/f8454d3f-dc7d-4ad2-afad-fcbfb0dec9f1)

Window stays open unless I close it manually:
![image](https://github.com/alangrainger/obsidian-google-photos/assets/516060/4dd16efc-e6d6-43ee-827d-a3684a944ca3)

**Comment by fbreitwieser** [comment]
I just checked with the 'Google Calendar' plugin where I manage to authenticate. There, the login window doesn't open in a Obsidian browser window, but rather my standalone browser (Chromium in that case). Once finished, the browser displays 'Authentication successful! Please return to obsidian.' (URL there is http://127.0.0.1:42813/callback...)

**Comment by alangrainger** [comment]
> I just checked with the 'Google Calendar' plugin where I manage to authenticate.

Excellent discovery! I will copy what they're doing there.

In the meantime (since that won't be today) do you have access to a Mac or Windows machine just for setting this up? All you need is to copy across the plugin's `data.json` file to your normal vault:

```
{
  "clientId": "",
  "clientSecret": "",
  "accessToken": ""
}
```

If you don't have access to a physical machine you could consider installing one in VirtualBox? Otherwise I will implement at some point the method from the Calendar plugin, since it sounds more reliable anyway and probably works on mobile, whereas BrowserWindow doesn't.

**Comment by fbreitwieser** [comment]
Thanks replying so quickly @alangrainger ! I'll use the workaround logging in on a Windows or Mac. Thank you :)

**Comment by Beingpax** [comment]
This doesn't work for me as well. The window is still stuck at this Approved clicked.

**Comment by alangrainger** [comment]
Thanks @Beingpax . I've started work implementing auth the same way as the Calendar plugin, hope to have something out next week.

**Comment by alangrainger** [comment]
Hi @fbreitwieser @Beingpax - this should hopefully be resolved. Can you please install `1.6.0` and test?

**Comment by alangrainger** [comment]
Hi @fbreitwieser @Beingpax - this should hopefully be resolved. Can you please install 1.6.0 and test?

**You will need to update the Redirect URI in your Photos API settings, as per the updated docs here:**

https://github.com/alangrainger/obsidian-google-photos/blob/main/docs/Setup.md

**Comment by fbreitwieser** [comment]
Thanks so much @alangrainger !! Just tested it out and it worked very well. (Minor issue was that a restart of Obsidian was required after the set-up, otherwise I'd get 'Error 400: redirect_uri_mismatches'). Thanks for your great work and plugin :)

**Comment by alangrainger** [comment]
Thanks - that restart issue is now fixed in 1.6.3
